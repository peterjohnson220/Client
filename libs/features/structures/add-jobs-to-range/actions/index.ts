export * from './jobs-to-grade.actions';
