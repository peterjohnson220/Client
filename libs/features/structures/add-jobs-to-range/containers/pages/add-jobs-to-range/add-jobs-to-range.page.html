<pf-search-layout *ngIf="pageShown$ | async" [resultsCount]="isSmallBiz ? null : (numberOfSearchResults$ | async)">
  <ng-container sidebar>

    <div *ngIf="addJobsConfig?.enableCreateNewJobs && !(searchingFilter$ | async)" [pfSecuredResource]="_Permissions.JOBS_ADD_EDIT_DELETE" class="create-new-job" [attr.data-qa-id]="'btn-createNewJobNav'">
      <div class="search-action" (click)="handleCreateNewJobClicked()">
        <fa-icon [icon]="['far', 'plus']"></fa-icon> Create New Job
      </div>
    </div>

    <pf-search-filters
      [defaultFocusedFilterId]="'jobTitleCode'"
      [visible]="!(searchingFilter$ | async)">
    </pf-search-filters>

    <pf-single-filter
      *ngIf="!!(searchingFilter$ | async)">
    </pf-single-filter>
  </ng-container>

  <ng-container content>

    <pf-results-header></pf-results-header>

    <pf-structures-jobs-search-results [canSelectJobs]="canSelectJobs"
                           [useSmallBizStyles]="isSmallBiz"
                           [customSearchResultsStyle]="!isJobRange ? {'width': '50%'} : {}"
                           [showJobBasedRangesJobMetadata]="addJobsConfig?.enableAssignedStructuresMetadata || false"
                           [showJobSourceOrTitle]="addJobsConfig?.enabledJobSourceOrTitle"
                           [isJobRange]="isJobRange"
                           [controlPoint]="controlPoint"
                           [gradeRangeGroupDetails]="gradeRangeGroupDetails">
    </pf-structures-jobs-search-results>

    <pf-paymarkets *ngIf="addJobsConfig?.enablePaymarkets" [useSmallBizStyles]="isSmallBiz"></pf-paymarkets>

    <pf-job-limit-counter *ngIf="!!isSmallBiz"
                          [jobUsageCount]="jobUsageCount"
                          [numberOfSearchResults]="(numberOfSearchResults$ | async)"
                          [currentJobCountSetting]="currentJobCountSetting"
                          [maxAllowedJobsSetting]="maxAllowedJobsSetting">
    </pf-job-limit-counter>

    <pf-jobs-to-grade-container *ngIf="!this.isJobRange"
      [rate]="rate"
      [gradeRangeGroupDetails]="gradeRangeGroupDetails"
      [controlPoint]="controlPoint">
    </pf-jobs-to-grade-container>

  </ng-container>

  <ng-container action-buttons>

    <span *ngIf="!(addingDataError$ | async)">
      <button class="btn btn-sm btn-secondary mr-3 clear-selections"
              [attr.data-qa-id]="'btn-clearSelections'"
              [disabled]="numberOfJobsSelected === 0"
              (click)="handleClearSelectionsClicked()">
        Clear Selections
      </button>

      <button *ngIf="isJobRange"
              class="btn btn-sm legacy-primary-btn mr-3 au-btn-add"
              [attr.data-qa-id]="'btn-addJobs'"
              [disabled]="(addingData$ | async)
                          || (numberOfJobsSelected === 0)
                          || (addJobsConfig?.enablePaymarkets && !(selectedPaymarkets$ | async)?.length)"
              (click)="handleAddClicked()">

        <ng-container *ngIf="!(addingData$ | async); else addingData">
          Add
          <span *ngIf="numberOfJobsSelected > 0">
            ({{ numberOfJobsSelected }})
          </span>
        </ng-container>
      </button>

      <button (click)="handleCancelClicked()"
              class="btn btn-sm btn-secondary mr-3 au-btn-cancel"
              [attr.data-qa-id]="'btn-cancelAddJobs'">
        Cancel
      </button>

      <button (click)="handleAddAllClicked()"
              class="btn btn-link mr-3 au-btn-addall"
              [attr.data-qa-id]="'btn-addAllJobs'"
              [disabled]="(addingData$ | async)
                          || (numberOfSearchResults$ | async) === 0"
              *ngIf="addJobsConfig?.enableAddAllJobs">
        <ng-container>
          Add All
          <span *ngIf="(numberOfSearchResults$ | async) > 0">
            {{ numberOfSearchResults$ | async }}
          </span>
        </ng-container>
      </button>
    </span>


    <span class="error-adding-jobs text-danger" *ngIf="!!(addingDataError$ | async)">
      {{
      (addingDataErrorMessage$ | async)
        ? (addingDataErrorMessage$ | async)
        : 'Error adding selected job(s).'
      }}
    </span>

  </ng-container>

</pf-search-layout>

<ng-template #addingData>
  Adding...
</ng-template>
