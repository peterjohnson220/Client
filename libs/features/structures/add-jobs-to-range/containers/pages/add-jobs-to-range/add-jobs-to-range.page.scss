@import '../../../../../search/search/styles/shared-variables';
@import '../../../../../search/search/styles/shared-styles';
//TODO: Create a non legacy version of this modal and apply these styles conditionally
@import '../../../../../../../assets/scss/legacy/legacy-styles';

.page {
  height: 100%;
}

.create-new-job{
  display: flex;
  align-items: center;
  margin-left: .5rem;
  height: $results-header-height;

  .fa-plus {
    margin-right: .15rem;
  }
}

:host ::ng-deep pf-search-filters .filters{
  @extend %position-below-results-header
}

:host ::ng-deep pf-search-results .results {
  width: 100%;
}

pf-paymarkets {
  @extend %position-below-results-header;
  width: 32.5% !important;
  right: 0;
}

:host ::ng-deep .content .actions-container {
  width: 100%;
}

.clear-selections{
  position: absolute;
  left: 0;
  margin-left: 1rem;
}

.auto-grade {
  position: absolute;
  left: 90px;
  bottom: 2px;
  margin-left: 1rem;
}

.error-adding-jobs {
  font-weight: bold;
}
