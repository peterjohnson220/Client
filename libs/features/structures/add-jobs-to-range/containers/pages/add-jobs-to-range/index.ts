export * from './add-jobs-to-range.page';
