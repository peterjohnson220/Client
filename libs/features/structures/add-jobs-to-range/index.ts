export * from './add-jobs-to-range.module';
