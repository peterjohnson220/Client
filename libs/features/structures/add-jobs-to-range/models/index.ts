export * from './grade.model';
export * from './grade-job.model';
export * from './grade-range-group-details.model';
export * from './get-add-grade-jobs.model';
export * from './grade-jobs.model';
