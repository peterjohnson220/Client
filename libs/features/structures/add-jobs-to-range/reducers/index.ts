import { createSelector, createFeatureSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromJobsToGradeReducer from './jobs-to-grade.reducer';

import { AddJobsState } from '../../../jobs/add-jobs/reducers';

// Feature area state
export interface AddJobsToRangeState extends AddJobsState {
  jobsToGrade: fromJobsToGradeReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  feature_addJobsToRange: AddJobsToRangeState;
}

// Feature area reducers
export const reducers = {
  jobsToGrade: fromJobsToGradeReducer.reducer
};

// Select Feature Area
export const selectFeatureAreaState = createFeatureSelector<AddJobsToRangeState>('feature_addJobsToRange');

// Feature Selectors
export const selectJobsToGradeState = createSelector(
  selectFeatureAreaState,
  (state: AddJobsToRangeState) => state.jobsToGrade
);

// Jobs to Grade Page Selectors


export const getLoadingGrades = createSelector(
  selectJobsToGradeState,
  fromJobsToGradeReducer.getLoadingGrades
);

export const getGrades = createSelector(
  selectJobsToGradeState,
  fromJobsToGradeReducer.getGrades
);

export const getLoadingGradesError = createSelector(
  selectJobsToGradeState,
  fromJobsToGradeReducer.getLoadingGradesError
);



