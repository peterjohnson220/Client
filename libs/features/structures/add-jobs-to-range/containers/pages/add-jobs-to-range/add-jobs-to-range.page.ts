import { Component, Input, OnDestroy, OnInit, Optional } from '@angular/core';
import { ActivatedRoute, Router } from '@angular/router';

import { Observable, Subscription } from 'rxjs';
import { Store } from '@ngrx/store';
import { DragulaService } from 'ng2-dragula';

import { Permissions, SystemUserGroupNames } from 'libs/constants';
import * as fromCompanySettingsActions from 'libs/state/app-context/actions/company-settings.actions';
import * as fromSearchFiltersActions from 'libs/features/search/search/actions/search-filters.actions';
import * as fromRootState from 'libs/state/state';
import { SettingsService } from 'libs/state/app-context/services';
import * as fromSearchReducer from 'libs/features/search/search/reducers';
import { SearchBaseDirective } from 'libs/features/search/search/containers/search-base';
import { UserContext } from 'libs/models/security';
import { CompanySettingsEnum } from 'libs/models/company';
import { AddJobsConfig, SearchFilterMappingData, staticFilters, JobSearchUserFilterType } from 'libs/features/jobs/add-jobs/data';
import * as fromPaymarketActions from 'libs/features/jobs/add-jobs/actions/paymarkets.actions';
import * as fromAddJobsReducer from 'libs/features/jobs/add-jobs/reducers';
import * as fromAddJobsPageActions from 'libs/features/jobs/add-jobs/actions/add-jobs-page.actions';
import * as fromAddJobsSearchResultsActions from 'libs/features/jobs/add-jobs/actions/search-results.actions';
import { SearchFeatureIds } from 'libs/features/search/search/enums/search-feature-ids';
import { ADD_JOBS_CONFIG_DEFAULT_TRUE } from 'libs/features/jobs/add-jobs/constants';

import { Grade, GradeRangeGroupDetails } from '../../../models';
import * as fromJobsToGradeActions from '../../../actions/jobs-to-grade.actions';
import * as fromJobsToGradeReducer from '../../../reducers';
import { cleanupDatacutsDragging, enableJobsDragging } from '../../../helpers';

@Component({
  selector: 'pf-add-jobs-to-range-page',
  templateUrl: './add-jobs-to-range.page.html',
  styleUrls: ['./add-jobs-to-range.page.scss']
})
export class AddJobsToRangePageComponent extends SearchBaseDirective implements OnInit, OnDestroy {
  @Input() isJobRange: boolean;
  @Input() controlPoint: string;
  @Input() rate: string;
  @Input() gradeRangeGroupDetails: GradeRangeGroupDetails;
  // Observables
  searchingFilter$: Observable<boolean>;
  numberOfSearchResults$: Observable<number>;
  jobCount$: Observable<number>;
  selectedPaymarkets$: Observable<number[]>;
  pageShown$: Observable<boolean>;
  addingData$: Observable<boolean>;
  addingDataError$: Observable<boolean>;
  addingDataErrorMessage$: Observable<any>;
  userContext: Observable<UserContext>;
  _Permissions = null;

  // Subscriptions
  selectedJobIdsSubscription: Subscription;
  selectedPayfactorsJobCodesSubscription: Subscription;
  gradesSubscription: Subscription;

  // Local variables
  maxAllowedJobsSetting = 0;
  currentJobCountSetting = 0;
  isSmallBiz = false;
  selectedJobIdCount: number;
  selectedJobCodeCount: number;
  grades: Grade[];

  // Add Jobs Config
  addJobsConfig: AddJobsConfig;
  constructor(
    store: Store<fromSearchReducer.State>,
    private router: Router,
    private route: ActivatedRoute,
    private settingsService: SettingsService,
    private dragulaService: DragulaService,
    @Optional() private injectedAddJobsConfig: AddJobsConfig

  ) {
    super(store, SearchFilterMappingData, SearchFeatureIds.AddJobs, JobSearchUserFilterType);
    this.addJobsConfig = injectedAddJobsConfig || ADD_JOBS_CONFIG_DEFAULT_TRUE;
    this.searchingFilter$ = this.store.select(fromSearchReducer.getSearchingFilter);
    this.numberOfSearchResults$ = this.store.select(fromSearchReducer.getNumberOfResultsOnServer);
    this.jobCount$ = this.store.select(fromAddJobsReducer.getJobCount);
    this.selectedPaymarkets$ = this.store.select(fromAddJobsReducer.getSelectedPaymarkets);
    this.pageShown$ = this.store.select(fromSearchReducer.getPageShown);
    this.addingData$ = this.store.select(fromAddJobsReducer.getAddingData);
    this.addingDataError$ = this.store.select(fromAddJobsReducer.getAddingDataError);
    this.addingDataErrorMessage$ = this.store.select(fromAddJobsReducer.getAddingDataErrorMessage);
    this.userContext = store.select(fromRootState.getUserContext);
    this._Permissions = Permissions;
    enableJobsDragging(dragulaService);
  }

  onSetContext(payload: any): void {
    this.store.dispatch(new fromAddJobsPageActions.SetContext(payload));
    this.store.dispatch(new fromSearchFiltersActions.AddFilters(staticFilters));
    this.store.dispatch(new fromPaymarketActions.SetDefaultPaymarket(Number(payload.PayMarketId)));
    this.store.dispatch(new fromPaymarketActions.LoadPaymarkets());


    // Always get the latest company settings so we have the latest job count
    this.store.dispatch(new fromCompanySettingsActions.LoadCompanySettings());
  }

  onResetApp(): void {
    this.store.dispatch(new fromPaymarketActions.ClearPayMarkets());
    this.store.dispatch(new fromAddJobsSearchResultsActions.ClearSelectedJobs());
  }

  handleAddClicked(): void {
    this.store.dispatch(new fromAddJobsPageActions.AddSelectedJobs());
  }

  handleSaveClicked(): void {
    this.store.dispatch(new fromJobsToGradeActions.SaveGradeJobMaps(this.grades.filter(g => g.JobIdsToAdd.length > 0 || g.JobIdsToRemove.length > 0)));
  }

  handleAddAllClicked(): void {
    this.store.dispatch(new fromAddJobsPageActions.AddAllJobs());
  }

  handleSelectAllClicked(): void {
    this.store.dispatch(new fromAddJobsSearchResultsActions.SelectAllJobs());
  }

  handleAutoGradeSelectedClicked(): void {
    // this.store.dispatch(new fromAddJobsSearchResultsActions.)
  }


  handleClearSelectionsClicked(): void {
    this.store.dispatch(new fromAddJobsSearchResultsActions.ClearSelectedJobs());
  }

  handleCreateNewJobClicked(): void {
    this.router.navigate(['../create-new-job'], {relativeTo: this.route});
  }

  get numberOfJobsSelected(): number {
    return (this.selectedJobIdCount + this.selectedJobCodeCount);
  }

  get jobUsageCount(): number {
    return this.currentJobCountSetting + this.numberOfJobsSelected;
  }

  get canSelectJobs(): boolean {
    if (!this.isSmallBiz) {
      return true;
    } else {
      return this.jobUsageCount < this.maxAllowedJobsSetting;
    }
  }

  disableSaveButton(): boolean {
    let disable = true;
    if (this.grades) {
      // see if there is at least one grade with jobs to remove or add, if yes, enable save button
      const found = this.grades.find(g => g.JobIdsToAdd.length > 0 || g.JobIdsToRemove.length > 0);
      disable = found == null;
    }
    return disable;
  }

  ngOnInit() {
    this.userContext.subscribe(uc => {
      this.isSmallBiz = (uc.CompanySystemUserGroupsGroupName === SystemUserGroupNames.SmallBusiness);
      if (this.isSmallBiz) {
        this.settingsService.selectCompanySetting<number>(CompanySettingsEnum.MaxProjectJobCount, 'number')
          .subscribe(setting => this.maxAllowedJobsSetting = setting);
        this.settingsService.selectCompanySetting<number>(CompanySettingsEnum.ProjectJobCount, 'number')
          .subscribe(setting => this.currentJobCountSetting = setting);
      }
    });

    this.selectedJobIdsSubscription = this.store.select(fromAddJobsReducer.getSelectedJobIds)
      .subscribe(jobIds => this.selectedJobIdCount = jobIds.length);
    this.selectedPayfactorsJobCodesSubscription = this.store.select(fromAddJobsReducer.getSelectedPayfactorsJobCodes)
      .subscribe(jobCodes => this.selectedJobCodeCount = jobCodes.length);
    this.gradesSubscription = this.store.select(fromJobsToGradeReducer.getGrades)
      .subscribe(grades => this.grades = grades);
  }

  ngOnDestroy() {
    this.selectedJobIdsSubscription?.unsubscribe();
    this.selectedPayfactorsJobCodesSubscription?.unsubscribe();
    this.gradesSubscription?.unsubscribe();
    cleanupDatacutsDragging(this.dragulaService);
  }

}
