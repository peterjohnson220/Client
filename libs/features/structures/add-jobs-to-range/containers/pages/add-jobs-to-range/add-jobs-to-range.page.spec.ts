import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';
// @ts-ignore
import { ActivatedRoute, Router } from '@angular/router';

import { NgbProgressbarModule } from '@ng-bootstrap/ng-bootstrap';
import { combineReducers, Store, StoreModule } from '@ngrx/store';
import { of } from 'rxjs';
import { DragulaModule } from 'ng2-dragula';

import * as fromRootState from 'libs/state/state';
import { generateMockUserContext } from 'libs/models/security';
import { SettingsService } from 'libs/state/app-context/services';
import * as fromPaymarketActions from 'libs/features/jobs/add-jobs/actions/paymarkets.actions';
import * as fromAddJobsPageActions from 'libs/features/jobs/add-jobs/actions/add-jobs-page.actions';
import * as fromAddJobsSearchResultsActions from 'libs/features/jobs/add-jobs/actions/search-results.actions';
import * as fromAddJobsReducer from 'libs/features/jobs/add-jobs/reducers';
import * as fromJobsToGradeActions from 'libs/features/structures/add-jobs-to-range/actions/jobs-to-grade.actions';

import { AddJobsToRangePageComponent } from './add-jobs-to-range.page';
import { generateMockGrade } from '../../../models';



describe('Project - Add Jobs - Structures Page', () => {
  let fixture: ComponentFixture<AddJobsToRangePageComponent>;
  let instance: AddJobsToRangePageComponent;
  let store: Store<fromAddJobsReducer.State>;
  let router: Router;
  let route: ActivatedRoute;

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          project_addJobs: combineReducers(fromAddJobsReducer.reducers),
        }),
        NgbProgressbarModule,
        DragulaModule.forRoot()
      ],
      declarations: [
        AddJobsToRangePageComponent
      ],
      providers: [
        {
          provide: ActivatedRoute,
          useValue: { url: 'fake-path/' }
        },
        {
          provide: Router,
          useValue: { navigate: jest.fn() },
        },
        {
          provide: SettingsService,
          useClass: SettingsService
        }
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    store = TestBed.inject(Store);
    router = TestBed.inject(Router);
    route = TestBed.inject(ActivatedRoute);

    fixture = TestBed.createComponent(AddJobsToRangePageComponent);
    instance = fixture.componentInstance;
  });

  it('should dispatch a set context action onSetContext', () => {
    const payload = {
      PayMarketId: 123,
      ProjectId: 1
    };
    const expectedAction = new fromAddJobsPageActions.SetContext(payload);

    jest.spyOn(store, 'dispatch');

    instance.onSetContext(payload);

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should dispatch a load paymarkets action onSetContext', () => {
    const payload = {
      PayMarketId: 123,
      ProjectId: 1
    };
    const expectedAction = new fromPaymarketActions.LoadPaymarkets();

    jest.spyOn(store, 'dispatch');

    instance.onSetContext(payload);

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should dispatch a set default paymarket action', () => {
    const payload = {
      PayMarketId: 123,
      ProjectId: 1
    };
    const expectedAction = new fromPaymarketActions.SetDefaultPaymarket(123);

    jest.spyOn(store, 'dispatch');

    instance.onSetContext(payload);

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should dispatch a clear paymarkets action onResetApp', () => {
    const expectedAction = new fromPaymarketActions.ClearPayMarkets();

    jest.spyOn(store, 'dispatch');

    instance.onResetApp();

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should dispatch a clear job search results action onResetApp', () => {
    const expectedAction = new fromAddJobsSearchResultsActions.ClearSelectedJobs();

    jest.spyOn(store, 'dispatch');

    instance.onResetApp();

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should dispatch a clear job search results action when handling clear selections clicked', () => {
    const expectedAction = new fromAddJobsSearchResultsActions.ClearSelectedJobs();

    jest.spyOn(store, 'dispatch');

    instance.handleClearSelectionsClicked();

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should dispatch a select all search results action when handling select all clicked', () => {
    const expectedAction = new fromAddJobsSearchResultsActions.SelectAllJobs();

    jest.spyOn(store, 'dispatch');

    instance.handleSelectAllClicked();

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should dispatch a save action when handling save clicked', () => {
    const grade = generateMockGrade();
    grade.JobIdsToAdd = [1];
    instance.grades = [grade];
    const expectedAction = new fromJobsToGradeActions.SaveGradeJobMaps(instance.grades);

    jest.spyOn(store, 'dispatch');

    instance.handleSaveClicked();

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should navigate to the create-new-job route relative to this one, when handling create new job clicked', () => {
    jest.spyOn(router, 'navigate');

    instance.handleCreateNewJobClicked();

    expect(router.navigate).toHaveBeenCalledWith(['../create-new-job'], { relativeTo: route });
  });

  it('should not display GUI features related to job count in non small biz companies, and should display the "normal"' +
    ' result count ', () => {
    instance.pageShown$ = of(true);
    instance.numberOfResults$ = of(100);
    instance.userContext = of(generateMockUserContext());

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
