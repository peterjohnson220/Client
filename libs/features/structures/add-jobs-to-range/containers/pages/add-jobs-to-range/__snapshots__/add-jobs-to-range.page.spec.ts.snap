// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project - Add Jobs - Structures Page should not display GUI features related to job count in non small biz companies, and should display the "normal" result count  1`] = `
<pf-add-jobs-to-range-page
  _Permissions={[Function Function]}
  addJobsConfig={[Function Object]}
  addingData$={[Function Store]}
  addingDataError$={[Function Store]}
  addingDataErrorMessage$={[Function Store]}
  currentJobCountSetting="0"
  dragulaService={[Function DragulaService]}
  gradesSubscription={[Function Subscriber]}
  injectedAddJobsConfig="null"
  isSmallBiz="false"
  jobCount$={[Function Store]}
  maxAllowedJobsSetting="0"
  numberOfResults$={[Function Observable]}
  numberOfSearchResults$={[Function Store]}
  pageShown$={[Function Observable]}
  route={[Function Object]}
  router={[Function Object]}
  searchFeatureId={[Function String]}
  searchFilterMappingDataObj={[Function Object]}
  searchingChildFilters$={[Function Store]}
  searchingFilter$={[Function Store]}
  selectedJobIdsSubscription={[Function Subscriber]}
  selectedPayfactorsJobCodesSubscription={[Function Subscriber]}
  selectedPaymarkets$={[Function Store]}
  settingsService={[Function SettingsService]}
  store={[Function Store]}
  userContext={[Function Observable]}
  userFilterTypeData={[Function Object]}
>
  <pf-search-layout>
    
    
    <div
      class="create-new-job"
      data-qa-id="btn-createNewJobNav"
    >
      <div
        class="search-action"
      >
        <fa-icon />
         Create New Job 
      </div>
    </div>
    <pf-search-filters />
    
    
    <pf-results-header />
    <pf-structures-jobs-search-results />
    
    <pf-paymarkets />
    
    
    <pf-jobs-to-grade-container />
    
    
    <span>
      
      <button
        class="btn btn-link mr-3 auto-grade"
        data-qa-id="btn-autoGradeSelectedJobs"
        disabled=""
      >
         Auto Grade Selected Jobs 
      </button>
      <button
        class="btn btn-sm btn-secondary mr-3 clear-selections"
        data-qa-id="btn-clearSelections"
      >
         Clear Selections 
      </button>
      
      
      <button
        class="btn btn-sm legacy-primary-btn mr-3 au-btn-add"
        data-qa-id="btn-saveJobs"
        disabled=""
      >
         Save 
      </button>
      <button
        class="btn btn-sm btn-secondary mr-3 au-btn-cancel"
        data-qa-id="btn-cancelAddJobs"
      >
         Cancel 
      </button>
      
      <button
        class="btn btn-link mr-3 au-btn-addall"
        data-qa-id="btn-selectAllJobs"
      >
        
         Select All 
        <span>
            
        </span>
      </button>
      
    </span>
    
  </pf-search-layout>
</pf-add-jobs-to-range-page>
`;
